cpu/sam0_common/periph/adc.c

```diff
@@ -177,17 +251,17 @@ int32_t adc_sample(adc_t line, adc_res_t res)
         return -1;
     }
 #ifdef CPU_SAMD21
-    ADC->INPUTCTRL.reg = ADC_GAIN_FACTOR_DEFAULT |
+    ADC_DEV->INPUTCTRL.reg = ADC_GAIN_FACTOR_DEFAULT |
```

Comment:
No gain on non-samd21 MCUs?

We could just 

```C
#ifndef ADC_GAIN_FACTOR_DEFAULT
#define ADC_GAIN_FACTOR_DEFAULT (0)
#endif
```

and remove the ifdef here.
Since 0 maps to `ADC_INPUTCTRL_GAIN_1X` that should be a sensible default.

Reply:
Should be sensible, yes. Done!